leptonai/cli/storage.py

```diff
@@ -297,26 +331,53 @@ def upload(local_path, remote_path, rsync, recursive, progress, file_system):
             flags += "a"
         if progress:
             flags += " --progress"
+        if partial or auto_resume:
+            flags += " --partial"
         command = (
             f"rsync {flags}"
             f" {local_path} rsync://{workspace_id}@{ip}:{port}/volume{remote_path}"
         )
-        console.print(f"Running command: [bold]{command}[/]")
-
-        process = subprocess.Popen(
-            command,
-            stdout=subprocess.PIPE,
-            stderr=subprocess.STDOUT,
-            env=env_vars,
-            shell=True,
-            universal_newlines=True,
-        )
 
-        for line in process.stdout:
-            print(line, end="")
-        process.wait()
+        MAX_RETRIES = 3
+        RETRY_DELAY = 5
+        attempts = MAX_RETRIES if auto_resume else 1
```

Comment:
I feel this is a bit confusing. We may want to have a single flag (not both ar and p) and blindly retry a couple of times

Reply:
Hi Cong,
Since --Partial only addresses network connectivity issues, I want the resume option to seamlessly manage similar scenarios without inconveniencing the user. I see your point that it might add unnecessary complexity. I don't want users to spend time figuring out the --partial option. Here’s what I’m thinking—let me know which one you prefer:

1. Enable retry by default for specific error codes (12, 20, 35, 23, 11, 10, 30, 5, 24) and provide a --no-recovery flag to disable it.
2. Use only --partial and let users manually retry.
3. Implement a standalone --auto-resume option for automatic recovery.

Reply:
I recommend choosing option 1, reducing the RETRY_DELAY time, and avoiding user notifications for the first 2-3 retries.

Reply:
I think we should do 3, while changing `--auto-resume` to a more user friendly name

Reply:
I think we should have a `--auto-retry-on-failure` flag (you can find a better name). Under the flag, we do `--partitial` and retry N times. Getting rid of all the complexities.